metrics/github/issues.py

```diff
@@ -0,0 +1,55 @@
+from collections import defaultdict
+from datetime import date, datetime, time
+
+import structlog
+
+from metrics.github import query, repos
+from metrics.tools.dates import iter_days
+
+
+log = structlog.get_logger()
+
+
+def get_metrics(client, orgs):
+    issues = get_issues(client, orgs)
+    log.info(f"Got {len(issues)} issues")
+
+    counts = calculate_counts(issues)
+    count_metrics = convert_to_metrics(counts)
+
+    return count_metrics
+
+
+def get_issues(client, orgs):
+    issues = []
+    for org in orgs:
+        for repo in repos.tech_repos(client, org):
+            issues.extend(query.issues(client, repo))
+    return issues
+
+
+def calculate_counts(issues):
+    counts = defaultdict(int)
+    for issue in issues:
+        start = issue.created_on
+        end = issue.closed_on if issue.closed_on else date.today()
+        for day in iter_days(start, end):
+            counts[(issue.repo.org, issue.repo.name, issue.author, day)] += 1
+    return dict(counts)
+
+
+def convert_to_metrics(counts):
+    metrics = []
+    for coord, count in counts.items():
+        org, repo, author, date_ = coord
```

Comment:
Just because I don't have anything else to talk about in this PR...

I generally avoid `from datetime import date` for exactly this reason. I either put up with monstrosities like `datetime.datetime.combine(...)` or, if I really can't live with the extra characters, then something like `import datetime as dt`.

And I suppose another option would be `from datetime import date as Date` – which is arguably what it should have been all along.

Reply:
I have a nostalgic affection for the lowercase stdlib type names generally, although that's not powerful enough to leave me with positive feelings about `datetime` specifically. Happy to go with fully-qualified names, especially since there's not a lot of datetimery in this module. Thanks.